surveys/static/js/charts.js

```diff
@@ -0,0 +1,336 @@
+// Make sure JQuery is loaded
+window.onload = function() {
+  if (!window.jQuery) {
+    throw new Error('JQuery must be loaded before charts.js');
+  }
+}
+
+var ChartHelper = function(surveys, types, bins) {
+  /**
+   * Helper class for displaying HighCharts charts on a page.
+   * This class depends on a number of nested objects as arguments in order to
+   * sensibly initialize chart display. For an example of the structure of these
+   * objects, see surveys.views.SurveySubmittedDetail.
+   * @constructor
+   * @param {Object} surveys - A data object containing survey results.
+   * @param {Object} types - An object of type definitions for survey results.
+   * @param {Object} bins - An object of bin definitions for binning survey results.
+   */
+  this.surveys = surveys;
+  this.countTypes = types.count;
+  this.observationalTypes = types.observational;
+  this.interceptTypes = types.intercept;
+  this.freeResponseInterceptTypes = types.freeResponseIntercept;
+  this.freeResponseInterceptBins = bins.freeResponseIntercept;
+}
+
+ChartHelper.prototype.loadChart = function(chartId, chartTitle, dataSourceId) {
+  /**
+   * Load a Highcharts chart.
+   * @param {String} chartId - The ID attribute of the chart container on the page.
+   * @param {String} chartTitle - The title of the chart to display.
+   * @param {String} dataSourceId - The ID of the primary data source to display.
+   */
+  // Check the type of the result to determine what kind of chart to display
+  var resultType;
+  if (this.surveys.length === 0) {
+    throw new Error('This survey has no collected data.');
+  } else {
+    resultType = this.surveys[0].data[dataSourceId].type;
+  }
+
+  var isCount = this.countTypes.indexOf(resultType) > -1;
+  var isObservational = this.observationalTypes.indexOf(resultType) > -1;
+  var isIntercept = this.interceptTypes.indexOf(resultType) > -1;
+  var isFreeResponseIntercept = this.freeResponseInterceptTypes.indexOf(resultType) > -1;
+
+  if (!(isCount || isObservational || isIntercept || isFreeResponseIntercept)) {
```

Comment:
A bullet from #97: Display a helpful error if there's only one run of the survey 

This check isn't happening as far as I can see. I'm not sure it's needed and feel fine about skipping it, but in later stages of this we should make it's easy to see how many surveys you're looking at in each chart.

Reply:
Thanks for catching that. I neglected to check my own spec!

I think showing the N for each survey is a great idea, and a reasonable ask for this PR. I'll go ahead and add it in.